test/metabase/models/util/spec_update_test.clj

```diff
@@ -0,0 +1,344 @@
+(ns metabase.models.util.spec-update-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.models.util.spec-update :as spec-update]
+   [toucan2.core :as t2]))
+
+(defn do-track-operations!
+  "Implementation function that tracks Toucan2 operations and returns the tracked operations."
+  [f]
+  (let [operations (atom [])
+        id-counter (atom 1)
+        track!     (fn [op args]
+                     (swap! operations conj (cons op args))
+                     1)]
+    (with-redefs [t2/insert!              (fn [& args] (track! :insert! args))
```

Comment:
With dynamic redefs?

Reply:
looks like with-dynamic-redefs is a bit buggy -- DMed you the bug.